src/com/stuypulse/stuylib/math/interpolation/Interpolator.java

```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2022 StuyPulse Robotics. All rights reserved. */
+/* This work is licensed under the terms of the MIT license */
+/* found in the root directory of this project. */
+package com.stuypulse.stuylib.math.interpolation;
+
+import java.util.Arrays;
+
+import com.stuypulse.stuylib.math.Vector2D;
+
+public interface Interpolator {
+    /**
+     *  a behavior that takes in a double and returns a double 
+     * @param x point to be intepolated
+     * @return interpolated value
+     */
+    double interpolate(double x); 
+
+
```

Comment:
add `default double get(double x) { return interpolate(x); }`

Reply:
exactly as written?

Reply:
should I remove `double interpolate(double x)`

Reply:
do not remove interpolate x 

the point of having another method called get(x) is to make an interpolator compatible with Ifilters. Ifilters require get x

Reply:
to make it compatible you also need to extent IFilter  tho so java knows that they are actually compatible